src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/GenericSecurityToken.cs

```diff
@@ -8,14 +8,14 @@
 
 namespace CoreWCF.IdentityModel.Tokens
 {
-    internal class GenericSecurityToken : SecurityToken
+    public class GenericSecurityToken : SecurityToken
     {
```

Comment:
I understand why you need to make this public, you need to create an instance of this in the Http project. As this is becoming public, it needs a little bit of design work. Can you rename it to GenericIdentitySecurityToken. Can you add a constructor which only takes a name and it auto generates the Id. Have a look at WindowsSecurityToken for an example. I'm also not sure how I feel about the expiration time being 10 hours out. Other token types which don't have an expiration time with them use SecurityUtils.MaxUtcDateTime. I think that's probably the most common sense option to use. Make it non-expiring.  
Also can you make the constructor take a GenericIdentity instead of the name? Constructing a fresh GenericIdentity instance from just the name loses a lot of data. It contains things like IsAuthenticated and AuthenticationType which we lose. There is a Clone() method on GenericIdentity so you can make a copy.